### DISCUSSION TITLE
Github token has no access to new container REST APIs

### DISCUSSION THREAD

**Discussion by kc1r74p** [community/community#26268] [Actions]
<p>Hello,</p>
<p>I just saw the changelog entry which now introduced access for the container registry.<br>
There it mentions that the secrets.GITHUB_TOKEN now has full access including reading and deleting for the ghcr registry.</p>
<p>However, switching from PAT to the GITHUB_TOKEN <strong>and adding permissions</strong> (push works fine with new token), now my custom action to list and then delete containers from <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a> throws:</p>
<blockquote>
<p>HttpError: Resource not accessible by integration</p>
</blockquote>
<p>when using following API with a non PAT token:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-the-authenticated-user" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-the-authenticated-user" target="_blank" rel="noopener nofollow ugc">Packages - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-the-authenticated-user" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/rest/reference/packages</a></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Could someone confirm or check this when using the GITHUB_TOKEN, did I miss something?<br>
Maybe PAT are still required when we want to use these APIs?<br>
It would be so nice if they also support the universal GITHUB_TOKEN :tada:</p>
<p>Best regards<br>
Patrick</p>

**Comment by kc1r74p** [comment]
<p>Related changelog post: <a href="https://github.blog/changelog/2021-03-24-packages-container-registry-now-supports-github_token/" class="inline-onebox" rel="noopener nofollow ugc">Packages: Container registry now supports GITHUB_TOKEN - GitHub Changelog</a></p>

**Comment by byrneh** [comment]
<aside class="quote no-group quote-modified" data-username="kc1r74p" data-post="1" data-topic="170395">
<div class="title">
<div class="quote-controls"></div>
 kc1r74p:</div>
<blockquote>
<p>However, switching from PAT to the GITHUB_TOKEN <strong>and adding permissions</strong> (push works fine with new token), now my custom action to list and then delete containers from <a href="http://ghcr.io" rel="noopener nofollow ugc">ghcr.io</a> throws:</p>
<blockquote>
<p>HttpError: Resource not accessible by integration</p>
</blockquote>
</blockquote>
</aside>
<p>Hi, <a class="mention" href="/kc1r74p">@kc1r74p</a>, welcome to the community.<br>
the API you linked to states<br>
Returns all package versions for a package owned <strong>by the authenticated user</strong>.<br>
<code>To use this endpoint, you must authenticate using an access token with the packages:read scope. If package_type is not container, your token must also include the repo scope.</code></p>
<p>The <code>GITHUB_TOKEN</code> secret is a GitHub App installation access token for a repository, it is not an authenticated user token, perhaps this explains the error message you see.</p>

**Comment by kc1r74p** [comment]
<p>Thanks <a class="mention" href="/byrneh">@byrneh</a>,<br>
yes I also thought that is most likely intentionally limited.</p>
<p>However which other API exists to read and delete containers if not those ?<br>
As the changelog mentions even delete, I was hoping that these APIs would be supported,<br>
as the only other alternative which I know of is to use the UI.</p>
<p>Best regards<br>
Patrick</p>

**Comment by byrneh** [comment]
<aside class="quote no-group" data-username="kc1r74p" data-post="4" data-topic="170395">
<div class="title">
<div class="quote-controls"></div>
 kc1r74p:</div>
<blockquote>
<p>However which other API exists to read and delete containers if not those ?<br>
As the changelog mentions even delete, I was hoping that these APIs would be supported,<br>
as the only other alternative which I know of is to use the UI.</p>
<p>Best regards<br>
Patrick</p>
</blockquote>
</aside>
<p>Hi,<br>
<a href="https://docs.github.com/en/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-a-user" rel="noopener nofollow ugc">packages#get-all-package-versions-for-a-package-owned-by-a-user</a> returns details for a user, rather than the authenticated user.<br>
Additionally, you will need to have have set appropriate access permissions<br>
<a href="https://docs.github.com/en/packages/guides/configuring-access-control-and-visibility-for-container-images" rel="noopener nofollow ugc">configuring-access-control-and-visibility-for-container-images</a></p>

**Comment by kc1r74p** [comment]
<aside class="quote no-group" data-username="byrneh" data-post="5" data-topic="170395">
<div class="title">
<div class="quote-controls"></div>
 byrneh:</div>
<blockquote>
<p>Hi,<br>
<a href="https://docs.github.com/en/rest/reference/packages#get-all-package-versions-for-a-package-owned-by-a-user" rel="noopener nofollow ugc">packages#get-all-package-versions-for-a-package-owned-by-a-user</a> returns details for a user, rather than the authenticated user.<br>
Additionally, you will need to have have set appropriate access permissions<br>
<a href="https://docs.github.com/en/packages/guides/configuring-access-control-and-visibility-for-container-images" rel="noopener nofollow ugc">configuring-access-control-and-visibility-for-container-images</a></p>
</blockquote>
</aside>
<p>I checked the user specific one as well, and it fails with the same message.<br>
I also rechecked the permissions, and pushing containers works fine with the GITHUB_TOKEN.<br>
So I am not seeing any other, more specific way to enable these APIs for the generic token.</p>
<p>Thanks none the less for checking!<br>
I will wait and see if there will be further information, I could imagine as it is beta, that it will just take some more time.</p>
<p>Best regards,<br>
Patrick</p>

**Comment by shayneburgess** [comment, marked as answer]
<p>For now, PATs are still required for the REST API. We plan to support GITHUB_TOKEN on the REST API in the future, however.</p>
<p>Shayne</p>

**Reply by thomastvedt** [reply]
Hello from the future, what's the status now? :)

**Reply by maxammann** [reply]
It's still not there, for GH Apps it returns an error  400 saying Invalid Argument.

Fine grained tokens are not supported at all.

Unfortunately the REST documentation is invalid by now. The endpoint for listing org packages claims to work with both.

**Reply by thomastvedt** [reply]
I see, thanks for the update :)

**Reply by countzero** [reply]
Is there a timeline for this feature?

**Comment by thomastvedt** [comment]
<p>Hi <a class="mention" href="/shayneburgess">@shayneburgess</a> , is there any progress here? 1 year later? Deleting a package via API and GITHUB_TOKEN seems like a pretty basic feature?</p>

**Comment by Yikun** [comment]
<p><a class="mention" href="/shayneburgess">@shayneburgess</a>, any progress? I think deleting a pakcage via API and GITHUB_TOKEN is very usefull for me.</p>

**Comment by maxammann** [comment]
The documentation says that apps and fine grained tokens are supported. No special permission should be required.

https://docs.github.com/en/rest/packages/packages?apiVersion=2022-11-28#list-packages-for-an-organization

It still fails with a 400 error code "Invalid argument"

**Reply by snath-bd** [reply]
Is this issue resolved? I am also getting this issue now from this api, when package_type=container, for all other package types, it returns result with success though. Seems like it is a bug. Can anyone shed light?

**Reply by maxammann** [reply]
might be worth it to go through gh support

**Comment by jr200** [comment]
Are there any updates on this - it would be great if GITHUB_TOKEN would work on this, or even App Installation Tokens..
